### PR TITLE
MM-1684 Refactored code to remove errors on startup

### PR DESCRIPTION
**Pull request by nickago**
Added keys to jsx elements within arrays to comply with react standards
Removed the 'async:false' clause from getMeSynchronous (now just getMe) since it was deprecated, not standard and not being used by any current functions.
Note, the middle commit is entirely just adding tabs, it had to be separated out to see the other cosmetic refactors easier.

### CODE REVIEW

File: web/react/stores/user_store.jsx [hunk 1]
```diff
@@ -18,244 +17,248 @@ var CHANGE_EVENT_STATUSES = 'change_statuses';
...
-      var me = client.getMeSynchronous();
-      if (me != null) {
-        this.setCurrentUser(me);
-        current_id = me.id;
-      }
-    }
+        return currentId;
+    },
+    getCurrentUser: function(skipFetch) {
+        if (this.getCurrentId(skipFetch) == null) {
```
Comment (ralder) [inline web/react/stores/user_store.jsx]: this still used parameters skipFetch that not exists already in getCurrentId

**Comment by nickago** [comment]
Good catch @ralder, I submitted a change to fix accordingly

**Comment by hmhealey** [comment]
Looks good to me

File: web/react/components/post_list.jsx [hunk 2]
```diff
@@ -439,9 +439,9 @@ module.exports = React.createClass({
                 currentPostDay = utils.getDateForUnixTicks(post.create_at);
                 if (currentPostDay.toDateString() != previousPostDay.toDateString()) {
                     postCtls.push(
-                        <div className="date-separator">
-                            <hr className="separator__hr" />
-                            <div className="separator__text">{currentPostDay.toDateString()}</div>
+                        <div key="date_div" className="date-separator">
```
Comment (ralder) [inline web/react/components/post_list.jsx]: it's not uniqe key when date separators are more then one. I shoud use something: `post.create_at` or  `currentPostDay.toDateString()`. And of course keys for all elements are not needed, only for dynamic children nodes

**Comment by crspeller** [comment]
@nickago Please fix @ralder's comment on unique keys.

**Comment by nickago** [comment]
@crspeller This should have fixed the problem now

File: web/react/components/post_list.jsx [hunk 3]
```diff
@@ -449,7 +449,7 @@ module.exports = React.createClass({
                 if (post.create_at > last_viewed && !rendered_last_viewed) {
                     rendered_last_viewed = true;
                     postCtls.push(
-                        <div className="new-separator">
+                        <div key="unviewed" className="new-separator">
```
Comment (crspeller) [inline web/react/components/post_list.jsx]: Is this key required?

File: web/react/components/sidebar_header.jsx [hunk 4]
```diff
@@ -87,7 +87,7 @@ var NavbarDropdown = React.createClass({
                 }
             });
         }
-        teams.push(<li><a href={utils.getWindowLocationOrigin() + '/signup_team'}>Create a New Team</a></li>);
+        teams.push(<li key='newTeam_li'><a key='newTeam_a' href={utils.getWindowLocationOrigin() + '/signup_team' }>Create a New Team</a></li>);
```
Comment (crspeller) [inline web/react/components/sidebar_header.jsx]: Is a key required here?

File: web/react/components/post_list.jsx (hunk 3 above)
Comment (ralder) [inline web/react/components/post_list.jsx]: yes

File: web/react/components/sidebar_header.jsx (hunk 4 above)
Comment (nickago) [inline web/react/components/sidebar_header.jsx]: Yes, anything that goes in an array needs a key

**Pull request merged**